src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -154,6 +155,35 @@ private static void ConfigureRegisteredBuilders<TBuilder>(TBuilder builder, ISer
             }
         }
 
+        private static void AddLockManagerIfAvailable(this IServiceCollection services, ScriptApplicationHostOptions applicationHostOptions)
+        {
+            try
+            {
+                IServiceProvider provider;
+                if (applicationHostOptions.HasParentScope)
+                {
+                    provider = applicationHostOptions.RootServiceProvider;
+                }
+                else
+                {
+                    provider = services.BuildServiceProvider();
```

Comment:
I might be missing some context here but building a temporary ServiceProvider is not a good idea in 98.5% of cases I've seen. Can we loop over the `services` and see if there are any `IAzureStorageProvider` registrations?

Reply:
So we need the actual service IAzureStorageProvider instantiated in order to check if Blob storage is available. We always register IAzureStorageProvider regardless, but here we don't want to add the BlobLeaseDistributedLockManager if blob storage isn't available (InMemoryDistributedLockManager already registered by webjobs).

Do you know what the concern is with building this ServiceProvider (i.e. any known issues)?

Reply:
>Do you know what the concern is with building this ServiceProvider (i.e. any known issues)?

Yeah, for one it's pretty expensive and might slow down the startup. The other is around resource management, the created service provider needs to be disposed.